tst/EnergyPlus/unit/ResultsFramework.unit.cc

```diff
@@ -457,43 +457,175 @@ TEST_F(ResultsFrameworkFixture, ResultsFramework_convertToMonth)
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.pushVariableValue(reportId, 3.0);
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.pushVariableValue(reportId, 4.0);
+
+    auto outputs = getCSVOutputs(*state,
+                                 state->dataResultsFramework->resultsFramework->RITimestepTSData.getJSON(),
+                                 *state->dataResultsFramework->resultsFramework,
+                                 OutputProcessor::ReportingFrequency::TimeStep);
+
+    std::map<std::string, std::vector<std::string>> expected_output = {
+        {"02/25 00:45:00", {"1.0"}}, {"02/25 01:00:00", {"2.0"}}, {"02/25 23:45:00", {"3.0"}}, {"02/25 24:00:00", {"4.0"}}};
+
+    EXPECT_EQ(expected_output, outputs);
+}
+
+TEST_F(ResultsFrameworkFixture, ResultsFramework_CSV_Timestamp_8601_End)
+{
+
+    json OutputData;
+    OutputProcessor::TimeStepType indexType = OutputProcessor::TimeStepType::Zone;
+    int reportId = 1;
+
+    Variable var0("SALESFLOOR INLET NODE:System Node Temperature", ReportingFrequency::TimeStep, indexType, reportId, Unit::C);
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.addVariable(var0);
+    state->dataResultsFramework->resultsFramework->addReportVariable(
+        "SALESFLOOR INLET NODE", "System Node Temperature", "C", ReportingFrequency::TimeStep);
+    state->dataResultsFramework->resultsFramework->setISO8601(true);
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.newRow(2, 25, 1, 45, 2017);  // month,day,hour,minute,year
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.newRow(2, 25, 1, 60, 2017);  // month,day,hour,minute,year
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.newRow(2, 25, 24, 45, 2017); // month,day,hour,minute,year
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.newRow(2, 25, 24, 60, 2017); // month,day,hour,minute,year
+
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.pushVariableValue(reportId, 1.0);
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.pushVariableValue(reportId, 2.0);
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.pushVariableValue(reportId, 3.0);
+    state->dataResultsFramework->resultsFramework->RITimestepTSData.pushVariableValue(reportId, 4.0);
+
+    auto outputs = getCSVOutputs(*state,
+                                 state->dataResultsFramework->resultsFramework->RITimestepTSData.getJSON(),
+                                 *state->dataResultsFramework->resultsFramework,
+                                 OutputProcessor::ReportingFrequency::TimeStep);
+
+    std::map<std::string, std::vector<std::string>> expected_output = {
+        {"2017-02-25T00:45:00", {"1.0"}}, {"2017-02-25T01:00:00", {"2.0"}}, {"2017-02-25T23:45:00", {"3.0"}}, {"2017-02-25T24:00:00", {"4.0"}}};
```

Comment:
ISO 8601 format with the traditional end of interval timestamps, so there's a `24:00:00` in the last timestamp.